client_impl.go

```diff
@@ -111,6 +112,7 @@ func (c *clientImpl) OnlineQueryBulk(params OnlineQueryParamsComplete) (OnlineQu
 			PreviewDeploymentId: params.underlying.PreviewDeploymentId,
 			Versioned:           params.underlying.versioned,
 			Branch:              params.underlying.BranchId,
+			Tags:                params.underlying.Tags,
```

Comment:
I think this method can definitely be confusing here - `sendRequestParams` as an object don't actually get sent over wholesale to the backend. `sendRequest` mainly cherry picks from these params and does different things with the params. For example, if `BranchId` is passed in, we'd include a branch Id header. And when `IsEngineRequest` is passed, we change the endpoint we hit. 

So basically passing in Tags here doesn't do anything

Reply:
the stuff that actually gets sent to the backend is `OnlineQueryRequest` which has the object `OnlineQueryContext` which has the field `Tags` - so I believe it's already being sent for online queries

Reply:
For offline and bulk queries we'd have to see in which spot to add the `Tags` field to the request, according to its corresponding server side object